### DISCUSSION TITLE
How much memory?

### DISCUSSION THREAD

**Discussion by 19Nazar** [pyodide/pyodide#4725] [Q&A]
Good afternoon, I have a question. I have a custom environment in webassemble that will provide such functions.
https://github.com/19Nazar/WasmZig
How much minimum memory will be required to run the assembly in python?

**Comment by ryanking13** [comment]
I think it depends on how many packages you use.

When I made some heap snapshot using browser dev tools (using Pyodide 0.25.1), the sizes were:

- After initializing Pyodide: ~50MB
- `import numpy`: ~100MB
- `import scipy`: ~300MB
- `import numpy, scipy, pandas, matplotlib, sklearn, matplotlib`: ~750MB

So I think the memory usage will vary greatly depending on how many packages you use and how many large files and C-extensions those packages consist of.

**Comment by 19Nazar** [comment]
I needed to know how much minimum memory I need, thank you very much, your answer helped a lot
